### PR TITLE
Reduce coupling to TelemetryLogger

### PR DESCRIPTION
**Pull request by anthony-murphy**
In a previous PR I deprecated the telemetry logger implementation. This change further decouple existing code from the TelemetryLogger, including a future change that will modify MockLogger to no longer inherit from TelemetryLogger, and only implement ITelemetryBaseLogger.

### CODE REVIEW

File: packages/utils/telemetry-utils/src/logger.ts [hunk 1]
```diff
@@ -238,14 +260,13 @@ export abstract class TelemetryLogger implements ITelemetryLoggerExt {
 						const value =
 							typeof getterOrValue === "function" ? getterOrValue() : getterOrValue;
 						if (value !== undefined) {
-							// @ts-expect-error what
```
Comment (anthony-murphy) [inline packages/utils/telemetry-utils/src/logger.ts:241]: this was a leftover artifact of the previous PR, #16385, this is just a bit of clean up to remove the expected error, which was spurious, just had to find typing to work around it.

**Review by alexvy86 — COMMENTED** [review]
What's the rationale for `ITelemetryLoggerExt` -> `ITelemetryBaseLogger`? When we pass loggers Internally it seems fine to be clear about "this one supports logging arrays/objects", whether the receiver leverages that or not.

**Comment by anthony-murphy** [comment]
> What's the rationale for `ITelemetryLoggerExt` -> `ITelemetryBaseLogger`? When we pass loggers Internally it seems fine to be clear about "this one supports logging arrays/objects", whether the receiver leverages that or not.

I'm deprecating TelemetryLogger, which in turn will make MockLogger a ITelemetryBaseLogger. These places all have unit tests that use MockLogger, so will break. To get around this we can wrap the MockLogger in a child logger to make it a ITelemetryLoggerExt, but in these cases that is also unnecessary as the cases already create a child logger with the passed in logger, so downgrading the type they take to ITelemetryBaseLogger will make the change to MockLogger transparent.

Overall, i think we should move away from passing around ITelemetryLoggerExt. I actually think we should make it internal. We will then pass around ITelemetryBaseLogger and create child loggers where necessary, and maybe add some internal helper functions for the few case were we pass ITelemetryLoggerExt into a func, only to call a single method on it, like sendError. Overall, this would make things much more consistent, e.g. always use ITelemetryBaseLogger, which also makes testing easier, as the interface is simpler.

File: packages/utils/telemetry-utils/src/logger.ts [hunk 2]
```diff
@@ -217,10 +238,11 @@ export abstract class TelemetryLogger implements ITelemetryLoggerExt {
 		return this.extendProperties(newEvent, includeErrorProps);
 	}
 
-	private extendProperties<T extends ITelemetryPropertiesExt>(
-		eventLike: T,
+	private extendProperties<T extends ITelemetryLoggerPropertyBag = ITelemetryLoggerPropertyBag>(
+		toExtend: T,
```
[the remark this answers is not in the document]
Comment (anthony-murphy) [reply packages/utils/telemetry-utils/src/logger.ts:242]: i didn't really change this code, i just moved it, it actually wasn't even necessary to move, it was an artifact of some other changes i was porting over from a different branch. if you look at prepare event, which is the only place this is called, and where it used to live, that code takes ITelemetryBaseEvent which extents ITelemetryLoggerPropertyBag, so i think this change is correct.

 I agree the typing is messed up here. It actually looks like a problem with ITelemetryPropertiesExt which probably should support getters, @markfields who should probably look into that.

[the remark this answers is not in the document]
Comment (anthony-murphy) [reply packages/utils/telemetry-utils/src/logger.ts:242]: i don't actually get ITelemetryPropertiesExt, or how that works with general object vs tagged objects, seems complicated, i'm also not sure i see the benefit to logging arrays and objects.

File: packages/runtime/container-runtime/src/summary/runningSummarizer.ts [hunk 3]
```diff
@@ -56,7 +56,7 @@ const defaultNumberSummarizationAttempts = 2; // only up to 2 attempts
  */
 export class RunningSummarizer implements IDisposable {
 	public static async start(
-		logger: ITelemetryLoggerExt,
```
Comment (markfields) [inline packages/runtime/container-runtime/src/summary/runningSummarizer.ts:59]: Feels like within the runtime we should be fine to pass around `ITelemetryLoggerExt`, why the change?

File: packages/utils/telemetry-utils/src/logger.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (markfields) [reply packages/utils/telemetry-utils/src/logger.ts:242]: Yeah the property bag types don't have Ext verions.  See https://dev.azure.com/fluidframework/internal/_queries/edit/4552, it's a leftover/low-pri todo from the work to add array support.

Tony, happy to discuss the "why" of supporting arrays and objects elsewhere

[the remark this answers is not in the document]
Comment (markfields) [reply packages/utils/telemetry-utils/src/logger.ts:242]: @anthony-murphy why isn't it just this?  There's only one caller and it passes an `ITelemetryBaseEvent`.

```suggestion
	private extendProperties(eventLike: ITelemetryBaseEvent,
```

File: packages/runtime/container-runtime/src/summary/runningSummarizer.ts (hunk 3 above)
Comment (markfields) [reply packages/runtime/container-runtime/src/summary/runningSummarizer.ts:59]: I found the existing convo, got it.

**Review by markfields — APPROVED** [review]

**Comment by anthony-murphy** [comment]
> > Overall, i think we should move away from passing around ITelemetryLoggerExt. I actually think we should make it internal. We will then pass around ITelemetryBaseLogger
> 
> Can you write up a one-pager and post on Dev and/or bring it to Design Review? Seems reasonable but would be good to discuss even briefly to check pros/cons

is there any documentation for how the current types work? it seems like a bit of a mess. my proposal is make it not a mess.

File: packages/utils/telemetry-utils/src/logger.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (anthony-murphy) [reply packages/utils/telemetry-utils/src/logger.ts:242]: its because i shouldn't have originally include it. i want to add a function to pull the properties off a logger to add to errors, which will reuse this, and has really been my end goal all along :)

**Comment by markfields** [comment]
@anthony-murphy let's revisit the "mess" after that ADO item about removing the old/duplicate types is done.
